src/test/java/core/basesyntax/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,163 @@
+package core.basesyntax;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.exception.RegistrationDataException;
+import core.basesyntax.model.User;
+import core.basesyntax.service.RegistrationService;
+import core.basesyntax.service.RegistrationServiceImpl;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static final String VALID_LOGIN = "validLogin";
+    private static final String VALID_PASSWORD = "validPassword";
+    private static final String EXISTING_LOGIN = "existingLogin";
+    private static final String ANOTHER_PASSWORD = "anotherPassword";
+    private static final String UNIQUE_LOGIN = "uniqueLogin";
+    private static final String UNIQUE_PASSWORD = "uniquePassword";
+    private static final String SHORT_LOGIN = "short";
+    private static final String SHORT_PASSWORD = "short";
+    private static final String EDGE_CASE_LENGTH = "length";
+    private static final int VALID_AGE = 20;
+    private static final int ANOTHER_AGE = 22;
+    private static final int UNIQUE_AGE = 25;
+    private static final int UNDERAGE = 17;
+
+    private RegistrationService registrationService;
+
+    @BeforeEach
+    void setUp() {
+        registrationService = new RegistrationServiceImpl();
+        Storage.people.clear();
+    }
+
+    @Test
+    void register_nullUser_notOk() {
+        Assertions.assertThrows(RegistrationDataException.class,
+                () -> registrationService.register(null));
+    }
+
+    @Test
+    void register_nullUserLogin_notOk() {
+        User user = new User();
+        user.setLogin(null);
+        user.setPassword(VALID_PASSWORD);
+        user.setAge(VALID_AGE);
```

Comment:
As Beata mentioned, you should extract user creation to a separate method.

![image](https://github.com/user-attachments/assets/7614dee5-38ae-4b6d-958c-df0840dfeb5f)